csrc/scheduler/utils.cpp

```diff
@@ -473,12 +473,38 @@ class PersistentBufferResolution : public IterVisitor {
 
 } // namespace
 
-// Returns all broadcast tvs from the given vector of Vals.
-std::vector<TensorView*> getBroadcastTvs(const std::vector<Val*>& dep_vals) {
+// Returns all broadcast tvs whose produers are reduction tvs and each broadcast
+// dim is mapped to a reduction dim.
+std::vector<TensorView*> getBufferProjectableBroadcastsTvs(
+    const std::vector<Val*>& dep_vals,
+    const TensorView* persistent_buffer) {
   std::vector<TensorView*> broadcast_tvs;
+  broadcast_tvs.reserve(dep_vals.size());
   for (auto val : dep_vals) {
     if (auto tv = dynamic_cast<TensorView*>(val)) {
-      if (tv->hasBroadcast()) {
+      if (!tv->definition()->isA<BroadcastOp>() || tv == persistent_buffer) {
+        continue;
+      }
+      const auto& producers = ir_utils::producerTvsOf(tv);
+      NVF_ERROR(
+          producers.size() == 1,
+          "Tv defined by BroadcastOp must have one produer! Got: ",
+          producers.size());
+      auto producer = producers.at(0);
+      if (!producer->definition()->isA<ReductionOp>()) {
+        continue;
+      }
+      // Each reduction dimension the producer, must be mapped to a broadcast
+      // dimension in the consumer, otherwise it is not a valid broadcast after
+      // reduction.
+      bool is_broadcast_after_reduction = true;
+      for (auto i : c10::irange(producer->nDims())) {
+        if (producer->axis(i)->isReduction() && !tv->axis(i)->isBroadcast()) {
```

Comment:
I don't think this is strictly correct as the number of dimensions may be different. Not common, but the normalization scheduler may accept a pattern like:

```
tv1 = sum(tv0, {0, 1});
tv2 = broadcast(tv1, {true});
tv3 = broadcast(tv2, {true, false});
```

Reply:
Good point. IN this case, we should project to `tv3`.  The revised algorithm solves this problem and a test case is added.